astropy/utils/tests/test_decorators.py

```diff
@@ -801,51 +815,58 @@ def docstring2():
     def testfunc3():
         """= 4 / 2 * test"""
 
-    assert inspect.getdoc(testfunc3) == "test + test = 4 / 2 * test"
+    expected_doc = "test + test = 4 / 2 * test" if sys.flags.optimize < 2 else None
+
+    assert inspect.getdoc(testfunc3) == expected_doc
 
 
 def test_format_doc_selfInput_simple():
     # Simple tests with self input
 
-    # Self input while the function has no docstring raises an error
-    with pytest.raises(ValueError):
+    if sys.flags.optimize < 2:
```

Comment:
Same comment as https://github.com/astropy/astropy/pull/17549/files#r1889354376 (wow didn't know this repeats so many times)